src/mpl/include/mpl_base.h

```diff
@@ -155,4 +155,16 @@ typedef bool _Bool;
 #define MPL_ROUND_UP_ALIGN(a, alignment) (((a) + ((alignment) - 1)) & (~((alignment) - 1)))
 #define MPL_ROUND_DOWN_ALIGN(a, alignment) ((a) & (~((alignment) - 1)))
 
+#define MPL_CAST(T, val) CAST_##T((val))
+#ifdef NDEBUG
+#define MPL_CAST_int(val) ((int) (val))
+#define MPL_CAST_Aint(val) ((MPI_Aint) (val))
+#else
+#define MPL_CAST_int(val) \
+    (((val) > INT_MAX || ((val) < 0 && (val) < INT_MIN)) ? (assert(0), 0) : (int) (val))
+#define MPL_CAST_Aint(val) \
+    (((val) > MPIR_AINT_MAX || ((val) < 0 && (val) < MPIR_AINT_MIN)) ? (assert(0), 0) \
```

Comment:
We cannot/shoudn't refer to MPIR macros in MPL.

Reply:
I moved the macro to `mpir_misc.h`. I think that should also be fine since we are not using `MPI_Aint` outside of MPI code.